src/components/PrimaryPanes/SourcesTree.js

```diff
@@ -91,6 +91,19 @@ class SourcesTree extends Component<Props, State> {
     });
   }
 
+  componentDidMount() {
+    const { selectedSource } = this.props;
+    if (selectedSource) {
+      const highlightItems = getDirectories(
+        selectedSource.get("url"),
+        this.state.sourceTree
+      );
+
+      // eslint-disable-next-line react/no-did-mount-set-state
+      this.setState({ highlightItems });
```

Comment:
Could we move this to the constructor?

Reply:
That was my first attempt but it didn't work

Reply:
"react/no-did-mount-set-state" is scaring me; let me think on this